core/gesture.js

```diff
@@ -316,27 +334,33 @@ Blockly.Gesture.prototype.startDraggingBlock_ = function() {
  * @param {!Event} e A mouse down or touch start event.
  */
 Blockly.Gesture.prototype.doStart = function(e) {
-  // TODO: Blockly.longStart_()
+  this.hasStarted_ = true;
+
   this.startWorkspace_.updateScreenCalculationsIfScrolled();
   this.startWorkspace_.markFocused();
   this.mostRecentEvent_ = e;
 
+  // Hide chaff also hides the flyout, so don't do it if the click is in a flyout.
   Blockly.hideChaff(!!this.flyout_);
 
+  if (this.startBlock_) {
+    this.startBlock_.select();
+  }
+
   if (Blockly.utils.isRightButton(e)) {
     this.handleRightClick(e);
     return;
   }
-  if (this.startBlock_) {
-    this.startBlock_.select();
+
+  if (goog.string.startsWith(e.type, 'touch')) {
```

Comment:
should this always be a "touchstart" event because it is called in doStart? or can a touchMove trigger a doStart as well?